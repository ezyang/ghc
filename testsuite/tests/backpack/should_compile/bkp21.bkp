unit p where
    signature H where
        data T

unit q where
    signature H where
        data T = T

unit pq0 where
    include p
    include q

unit pq1 where
    include p
    include q
    signature H where
        data T = T

unit pq2 where
    include p
    include q
    signature H where
        data T
