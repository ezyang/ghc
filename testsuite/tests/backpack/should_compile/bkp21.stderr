[1 of 5] Processing p
/--- Shape for p
provides:
requires: H -> T{T}
\---
  [1 of 1] Compiling H[sig]           ( p/H.hsig, nothing )
[2 of 5] Processing q
/--- Shape for q
provides:
requires: H -> T{T, T}
\---
  [1 of 1] Compiling H[sig]           ( q/H.hsig, nothing )
[3 of 5] Processing pq0
/--- Shape for pq0
provides:
requires: H -> T{T, T}
\---
  [1 of 1] Compiling H[sig]           ( pq0/H.hsig, nothing )
[4 of 5] Processing pq1
/--- Shape for pq1
provides:
requires: H -> T{T, T}
\---
  [1 of 1] Compiling H[sig]           ( pq1/H.hsig, nothing )
[5 of 5] Processing pq2
/--- Shape for pq2
provides:
requires: H -> T{T, T}
\---
  [1 of 1] Compiling H[sig]           ( pq2/H.hsig, nothing )
